data-model-layer/data-model.md

```diff
@@ -85,9 +85,42 @@ Boolean is a scalar kind.  Its cardinality is two -- either the value 'true' or
 
 #### Integer kind
 
+Integer is a scalar kind and refers to whole numbers without a fractional
+component.
+
+It is important to consider codec and language limitations that may be imposed
+on the serialization of integers from the IPLD Data Model. For example:
+
+* Some codecs, such as DAG-CBOR, will assume that integers must be within the
+  64-bit signed range and reject anything larger.
+* IPLD libraries, such as go-ipld-prime, limit their in-memory representation
+  of the integer kind to the signed 64-bit range.
+* JavaScript has difficulties safely handling and representing integers outside
+  of the 53-bit unsigned range and differentiating between integers and floats
+	where there is no fractional component.
 
 #### Float kind
 
+Float is a scalar kind and refers to a number with a fractional component,
+represented with a decimal point. This may also include numbers where the
+fractional component is zero (although the distinction between these numbers
+and integers is blurred in some cases—specifically the JavaScript memory model).
+
+While Float is a formal kind in the IPLD Data Model, **it is recommended that
+Float values be avoided when developing systems on IPLD** (and
+content-addressable systems in general) due to the broad scope for introducing
+variability in byte representations. Alternative methods for encoding
+fractional numbers with greater precision and less variability should be
+considered where possible.
+
+There is not a strict 1:1 mapping between the Float kind and any particular
```

Comment:
But we still do support a subset of IEEE 754 though, right? Otherwise it almost seems like Float is little more than a bunch of arbitrary bytes, which might contain practically anything from a spec point of view.

Reply:
Yes, this is a tricky point - we're talking about the data model here which is about what you interact with programmatically, so (1) that's a step away from codecs, where the rubber hits the road on the bytes->something question, and (2) this is a matter for programming languages and IPLD libraries.

I've added this to try and get closer to the point:

> However, for practical purposes, the IPLD Float kind will be implemented using IEEE 754 primitives and byte layout in most languages and codecs due to its ubiquity.

I hope that helps clarify, along with some other words I've added.

Reply:
Yeah, I think that wording does help.